gsd/gsd.c

```diff
@@ -31,37 +35,98 @@
 // define windows wrapper functions
 #ifdef _WIN32
 #define lseek _lseeki64
-#define write _write
-#define read _read
 #define open _open
 #define ftruncate _chsize
 #define fsync _commit
+typedef int64_t ssize_t;
 
 int S_IRUSR = _S_IREAD;
 int S_IWUSR = _S_IWRITE;
 int S_IRGRP = _S_IREAD;
 int S_IWGRP = _S_IWRITE;
 
-size_t pread(int fd, void *buf, size_t count, int64_t offset)
+ssize_t pread(int fd, void *buf, size_t count, int64_t offset)
     {
+    // Note: _read only accepts unsigned int values
+    if (count > UINT_MAX)
+        return -1;
+
     int64_t oldpos = _telli64(fd);
     _lseeki64(fd, offset, SEEK_SET);
-    size_t result = _read(fd, buf, count);  // Note: does not support >4GB reads
+    ssize_t result = _read(fd, buf, (unsigned int)count);
     _lseeki64(fd, oldpos, SEEK_SET);
     return result;
     }
 
-size_t pwrite(int fd, void *buf, size_t count, int64_t offset)
+ssize_t pwrite(int fd, const void *buf, size_t count, int64_t offset)
     {
+    // Note: _write only accepts unsigned int values
+    if (count > UINT_MAX)
+        return -1;
+
     int64_t oldpos = _telli64(fd);
     _lseeki64(fd, offset, SEEK_SET);
-    size_t result = _write(fd, buf, count);  // Note: does not support >4GB writes
+    ssize_t result = _write(fd, buf, (unsigned int)count);
     _lseeki64(fd, oldpos, SEEK_SET);
     return result;
     }
 
 #endif
 
+static ssize_t __pwrite_retry(int fd, const void *buf, size_t count, int64_t offset)
+    {
+    size_t total_bytes_written = 0;
```

Comment:
Should this be `ssize_t`?

Reply:
I don't think so, the total count of bytes written will never be negative.

Reply:
Oh I see, I was misinterpreting the difference between `ssize_t`  and `size_t` as memory size, but it's just the sign.